code/src/nuvla/connector/kubernetes.py

```diff
@@ -169,15 +169,84 @@ def list(self, filters=None):
+                                 '--namespace', namespace] + since_opt
+                container_opts = \
+                ['pod/' + pod_unique_id, '--container=' + container]
+                cmd = \
+                self.build_cmd_line(['logs'] + container_opts + list_opts_log)
+                log.debug('Generated logs command line : %s', cmd)
+                header_line = "\n\nLog last " + str(tail_lines) + \
+                " lines for Container " + \
+                container + " in Pod " + pod_unique_id + " \n"
+                log.debug('Header line : %s', header_line)
+                logs_string = \
+                logs_string + header_line + execute_cmd(cmd).stdout
+                log.debug('_get_container_logs logs string : %s', logs_string)
+        except Exception as e_cont:
+            ex_string = "There was a problem getting logs from Pod " \
+            + pod_unique_id
+            log.debug('An exception caught: %s ', e_cont)
+            logs_string = str(ex_string)
+        return logs_string 
+
+    def _get_the_pods(self, namespace, values, since_opt, lines: int) -> str:
+        logs_string = str()
+        log.debug('Starting _get_containers_logs.')
+        for items_list in values['items']:
+            if str(items_list["kind"]) == 'Pod':
+                logs_string = logs_string + \
+                self._get_container_logs(namespace, items_list, since_opt, lines)
+        log.debug('_get_containers FINAL log string : %s', logs_string)
+
+        return logs_string
+
+    def _get_the_logs(self, namespace, since_opt, lines: int) -> str:
+        list_opts_pods = ['-o', 'json', '--namespace', namespace]
+        cmd_pods = self.build_cmd_line(['get', 'pods'] + list_opts_pods)
+        log.debug('Generated command line to get pods: %s', cmd_pods)
+        try:
+            all_json_out = json.loads(execute_cmd(cmd_pods).stdout)
+            try:
+                log.info('Getting containers...')
+                logs_string = \
+                self._get_the_pods(namespace, all_json_out, since_opt, lines)
+            except Exception as e_cont:
+                log.debug('Fetching Containers failed: %s ', e_cont)
+        except Exception as e_json:
+            log.debug('Fetching JSON failed: %s', e_json)
+        # FIXME I leave for now
+        log.info('Container logs string: %s', logs_string)
+        return logs_string
+
     @should_connect
     def log(self, component: str, since: datetime, lines: int,
             **kwargs) -> str:
         namespace = kwargs['namespace']
         since_opt = ['--since-time', since.isoformat()] if since else []
-        list_opts = [component, '--timestamps=true', '--tail', str(lines),
-                     '--namespace', namespace] + since_opt
-        cmd = self.build_cmd_line(['logs'] + list_opts)
-        return execute_cmd(cmd).stdout
+        do_not_send_logs = ["Service"]
+        if component.split("/")[0] not in do_not_send_logs:
```

Comment:
Can be simplified and improved.

```
WORKLOAD_OBJECT_KINDS = ['Deploymet', 'Job', 'CronJob', 'StatefulSet', 'DaemonSet']

if component.split("/")[0] not in WORKLOAD_OBJECT_KINDS:
    msg = f'Logs can not be collected for {component} object kind.'
    log.info(msg)
    return msg + '\n'

try:
    log.debug('Getting container logs for %s', component)
    return self._get_the_logs(namespace, since_opt, lines)
except Exception as ex:
    log.error('Failed getting container logs for %s: %s', component, ex)
    return '' # if called can't handle None
```

Reply:
We can do this type of logic but must return a string in the form of 
`2023-06-26T12:51:29.676109000Z <any text>`
Otherwise the Clojure template check fails...

Reply:
The `log` function as described above is implemented now.